data-search-widget/data-search-widget.js

```diff
@@ -1294,11 +1297,59 @@
         })
         .addTo(map);
     addMarkers(mapsData);
+    $('button[type=submit]').on('click', function () {
+      includeMobileResults = true;
+      // This is to display all the results even if they are missing lat and long
+    })
+    map.addEventListener('moveend', mapMoveCallback);
   }
 
+  function mapMoveCallback(event) {
+    if (!includeMobileResults) {
+      clearMarkers();
+      globalSearchTool.template.paginate(getInBoundResults());
+    } else {
+      includeMobileResults = false;
+    }
+  }
+
+  function getInBoundResults() {
+    let markers = {};
+    let gridSize = 30;
+    var results = []
+    let markerClusters = L.markerClusterGroup({
+      showCoverageOnHover: false,
+      animateAddingMarkers: true,
+      maxClusterRadius: gridSize,
+    });
+    $.each(lastFilteredResults, function (key, item) {
+      if (item.latitude != "(blank)" && item.longitude != "(blank)") {
+        // put it on the map?
+        if (!item.latitude) {
+          return;
+        }
+        var thisLatlng = L.latLng(item.latitude, item.longitude);
+        if (map.getBounds().contains(thisLatlng)) {
+          var latlong = item.latitude + ',' + item.longitude;
+          if (!markers[latlong]) {
+            // add marker to map
+            markers[latlong] = L.marker(new L.LatLng(item.latitude, item.longitude));
+            markers[latlong].bindPopup(globalConfig.resultTemplate.markerPopupText(item));
+            markerClusters.addLayer(markers[latlong]);
+            results.push(item);
+          }
+        }
+      }
+    });
+    //markerClusters.addTo(map);
+    map.addLayer(markerClusters);
+    globalClusters = markerClusters;
+    return results;
+  }
   function addMarkers(mapsData) {
```

Comment:
```suggestion
  function addMarkers(mapsData, updateMapBounds) {
```
Optional `updateMap` argument to control whether map bounds are updated.